lib/componentRelay.ts

```diff
@@ -217,6 +206,17 @@ export default class ComponentRelay {
         return
       }
 
+      /**
+       * The Component Registered message will be the most reliable one, so we won't change it after any subsequent events,
+       * in case you receive an event from another window.
+       */
+      if (parsedData.action === ComponentAction.ComponentRegistered) {
```

Comment:
We should also check to make sure the current origin is undefined. Otherwise a malicious component can just resend the ComponentRegistered event arbitrarily many times.

Reply:
```suggestion
      if (typeof this.component.origin === 'undefined' && parsedData.action === ComponentAction.ComponentRegistered) {
```

Reply:
```suggestion
      if (!this.component.origin && parsedData.action === ComponentAction.ComponentRegistered) {
```

Reply:
Please test one more time with this new change and let us know if it still works as expected.

Reply:
> We should also check to make sure the current origin is undefined. Otherwise a malicious component can just resend the ComponentRegistered event arbitrarily many times.

`!this.component.origin` will always return true on iOS because `!"" = true`. Correct me if I am wrong, but this would allow the first `ComponentAction.ComponentRegistered` to be from SN and set the origin to `""` which we want. Then, the second `ComponentAction.ComponentRegistered` message could be malicious and set the origin to an unknown value. 

With my proposed change, `typeof this.component.origin === 'undefined'`, any future `ComponentAction.ComponentRegistered` would be ignored.

Reply:
Hmm, interesting that `!""` does indeed resolve to true. That's unexpected. In that case your solution would be better.

Reply:
I just waver over all the different ways to check for undefined and not sure of their equivalence.

`value === null || value === undefined || typeof value === 'undefined'`

Reply:
> I just waver over all the different ways to check for undefined and not sure of their equivalence.
`value === null || value === undefined || typeof value === 'undefined'`

I'm not sure if this was a question or not, but if it was this is my understanding of the differences: 

`value === null` has to be set, so it usually shows the intent that you want a variable to be empty.
`value === undefined` just means a value has not been assigned to the variable yet.
`typeof value === 'undefined'` is a safer version of checking `value === undefined` because it will not throw an error if a variable has not yet been created.

```js
let value;

// Same results
if (value === undefined) // Returns true
(typeof value === 'undefined') // Returns true

// Notice value2 has not been set
if (value2 === undefined) // throws error
if (typeof value2 === 'undefined') // returns true
```

> Hmm, interesting that !"" does indeed resolve to true. That's unexpected.

Agreed, it is quite unexpected.

I will test all the changes and see if everything still works.